### PR TITLE
Updated dependencies to work with forge 3.8.1 and swarm 2018.3.3

### PR DESCRIPTION
**Pull request by mmicheo**
These changes were necessary to be able to add the wildfly-swarm-addon to forge.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -8,13 +8,13 @@
    <modelVersion>4.0.0</modelVersion>
    <groupId>org.jboss.forge.addon</groupId>
    <artifactId>wildfly-swarm</artifactId>
-   <version>2017.12.0-SNAPSHOT</version>
+   <version>2017.11.0</version>
```
Comment (gastaldi) [inline pom.xml]: The version should still be SNAPSHOT. I'll perform a mvn release once the deps are ok

Comment (mmicheo) [reply pom.xml]: Thanks!

**Comment by gastaldi** [comment]
[test]

**Comment by gastaldi** [comment]
Retest this please

**Comment by gastaldi** [comment]
Thanks!

**Comment by mmicheo** [comment]
Errors when installing addon from https://github.com/forge/wildfly-swarm-addon.git
[attempt-from-forge.log](https://github.com/forge/wildfly-swarm-addon/files/1804038/attempt-from-forge.log)

Errors when installing addon from https://github.com/mmicheo/wildfly-swarm-addon.git (with SNAPSHOT)
[attempt-from-fork-with-SNAPSHOT.txt](https://github.com/forge/wildfly-swarm-addon/files/1804039/attempt-from-fork-with-SNAPSHOT.txt)

No errors when installing addon from https://github.com/mmicheo/wildfly-swarm-addon.git (without SNAPSHOT)
[attempt-from-fork-without-SNAPSHOT.txt](https://github.com/forge/wildfly-swarm-addon/files/1804040/attempt-from-fork-without-SNAPSHOT.txt)

**Comment by gastaldi** [comment]
```bash
$ addon-install-from-catalog --addon org.jboss.forge.addon:wildfly-swarm 
addon-install-from-git --url https://github.com/forge/wildfly-swarm-addon.git --coordinate org.jboss.forge.addon:wildfly-swarm
***INFO*** Installing Addon from Git 	[1/4] ...
***INFO*** Installing Addon from Git:Cloning repository in /tmp/tmpdir6414122938675388712 	[1/4] ...
***INFO*** Installing Addon from Git:Installing project into local repository 	[2/4] ...
[INFO] Scanning for projects...
[INFO] Inspecting build with total of 1 modules...
[INFO] Installing Nexus Staging features:
[INFO]   ... total of 1 executions of maven-deploy-plugin replaced with nexus-staging-maven-plugin
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] Building wildfly-swarm 2018.4.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:3.0.0:clean (default-clean) @ wildfly-swarm ---
[INFO] 
[INFO] --- maven-enforcer-plugin:1.4.1:enforce (enforce-java-version) @ wildfly-swarm ---
[INFO] 
[INFO] --- maven-enforcer-plugin:1.4.1:enforce (enforce-maven-version) @ wildfly-swarm ---
[INFO] 
[INFO] --- buildnumber-maven-plugin:1.4:create (get-scm-revision) @ wildfly-swarm ---
[INFO] Executing: /bin/sh -c cd '/tmp/tmpdir6414122938675388712' && 'git' 'rev-parse' '--verify' 'HEAD'
[INFO] Working directory: /tmp/tmpdir6414122938675388712
[INFO] Storing buildNumber: d9c35289fae322953a8ef1de86a4183a5ba363d1 at timestamp: 1520879265015
[INFO] Storing buildScmBranch: master
[INFO] 
[INFO] --- maven-resources-plugin:3.0.1:resources (default-resources) @ wildfly-swarm ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 1 resource
[INFO] 
[INFO] --- maven-compiler-plugin:3.6.0:compile (default-compile) @ wildfly-swarm ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 14 source files to /tmp/tmpdir6414122938675388712/target/classes
[INFO] 
[INFO] --- maven-resources-plugin:3.0.1:testResources (default-testResources) @ wildfly-swarm ---
[INFO] Not copying test resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.6.0:testCompile (default-testCompile) @ wildfly-swarm ---
[INFO] Not compiling test sources
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 4 source files to /tmp/tmpdir6414122938675388712/target/test-classes
[INFO] 
[INFO] --- maven-surefire-plugin:2.19.1:test (default-test) @ wildfly-swarm ---
[INFO] Tests are skipped.
[INFO] 
[INFO] --- maven-jar-plugin:3.0.2:jar (default-jar) @ wildfly-swarm ---
[INFO] Building jar: /tmp/tmpdir6414122938675388712/target/wildfly-swarm-addon.jar
[INFO] 
[INFO] --- maven-source-plugin:3.0.1:jar-no-fork (attach-sources) @ wildfly-swarm ---
[INFO] Building jar: /tmp/tmpdir6414122938675388712/target/wildfly-swarm-addon-sources.jar
[INFO] 
[INFO] --- maven-jar-plugin:3.0.2:jar (create-forge-addon) @ wildfly-swarm ---
[INFO] Building jar: /tmp/tmpdir6414122938675388712/target/wildfly-swarm-addon-forge-addon.jar
[INFO] 
[INFO] --- maven-install-plugin:2.5.2:install (default-install) @ wildfly-swarm ---
[INFO] Installing /tmp/tmpdir6414122938675388712/target/wildfly-swarm-addon.jar to /home/ggastald/.m2/repository/org/jboss/forge/addon/wildfly-swarm/2018.4.0-SNAPSHOT/wildfly-swarm-2018.4.0-SNAPSHOT.jar
[INFO] Installing /tmp/tmpdir6414122938675388712/pom.xml to /home/ggastald/.m2/repository/org/jboss/forge/addon/wildfly-swarm/2018.4.0-SNAPSHOT/wildfly-swarm-2018.4.0-SNAPSHOT.pom
[INFO] Installing /tmp/tmpdir6414122938675388712/target/wildfly-swarm-addon-sources.jar to /home/ggastald/.m2/repository/org/jboss/forge/addon/wildfly-swarm/2018.4.0-SNAPSHOT/wildfly-swarm-2018.4.0-SNAPSHOT-sources.jar
[INFO] Installing /tmp/tmpdir6414122938675388712/target/wildfly-swarm-addon-forge-addon.jar to /home/ggastald/.m2/repository/org/jboss/forge/addon/wildfly-swarm/2018.4.0-SNAPSHOT/wildfly-swarm-2018.4.0-SNAPSHOT-forge-addon.jar
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 5.693 s
[INFO] Finished at: 2018-03-12T15:27:48-03:00
[INFO] Final Memory: 78M/893M
[INFO] ------------------------------------------------------------------------
***INFO*** Installing Addon from Git:Removing previous addon installation (org.jboss.forge.addon:wildfly-swarm,2018.4.0-SNAPSHOT) 	[3/4] ...
***INFO*** Installing Addon from Git:Installing addon (org.jboss.forge.addon:wildfly-swarm,2018.4.0-SNAPSHOT) 	[4/4] ...
***SUCCESS*** Addon(s) [org.jboss.forge.addon:wildfly-swarm,2018.4.0-SNAPSHOT] installed successfully.
[launchpad]$ ***SUCCESS*** Success [Addon(s) [org.jboss.forge.addon:wildfly-swarm,2018.4.0-SNAPSHOT] installed successfully.]
```

**Comment by gastaldi** [comment]
See if the errors are gone if you remove everything in your `~/.forge` directory

**Comment by mmicheo** [comment]
That worked, thank you!

❯ forge
Using Forge at /usr/local/Cellar/jboss-forge/3.8.1.Final/libexec

    _____
   |  ___|__  _ __ __ _  ___
   | |_ / _ \| `__/ _` |/ _ \  \\
   |  _| (_) | | | (_| |  __/  //
   |_|  \___/|_|  \__, |\___|
                   |__/

JBoss Forge, version [ 3.8.1.Final ] - JBoss, by Red Hat, Inc. [ http://forge.jboss.org ]
Hit '<TAB>' for a list of available commands and 'man [cmd]' for help on a specific command.

To quit the shell, type 'exit'.

[micheom]$ addon-install-from-git --url https://github.com/forge/wildfly-swarm-addon.git
***INFO*** Installing Addon from Git 	[1/4] ...
***INFO*** Installing Addon from Git:Cloning repository in /var/folders/dd/gx1cd7ws7wq8tf6phrjxqbmsnnzm5j/T/tmpdir1109678367441608430 	[1/4] ...
***INFO*** Installing Addon from Git:Installing project into local repository 	[2/4] ...
[INFO] Scanning for projects...
[INFO] Inspecting build with total of 1 modules...
[INFO] Installing Nexus Staging features:
[INFO]   ... total of 1 executions of maven-deploy-plugin replaced with nexus-staging-maven-plugin
[INFO]
[INFO] ------------------------------------------------------------------------
[INFO] Building wildfly-swarm 2018.4.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO]
[INFO] --- maven-clean-plugin:3.0.0:clean (default-clean) @ wildfly-swarm ---
[INFO]
[INFO] --- maven-enforcer-plugin:1.4.1:enforce (enforce-java-version) @ wildfly-swarm ---
[INFO]
[INFO] --- maven-enforcer-plugin:1.4.1:enforce (enforce-maven-version) @ wildfly-swarm ---
[INFO]
[INFO] --- buildnumber-maven-plugin:1.4:create (get-scm-revision) @ wildfly-swarm ---
[INFO] Executing: /bin/sh -c cd '/var/folders/dd/gx1cd7ws7wq8tf6phrjxqbmsnnzm5j/T/tmpdir1109678367441608430' && 'git' 'rev-parse' '--verify' 'HEAD'
[INFO] Working directory: /var/folders/dd/gx1cd7ws7wq8tf6phrjxqbmsnnzm5j/T/tmpdir1109678367441608430
[INFO] Storing buildNumber: d9c35289fae322953a8ef1de86a4183a5ba363d1 at timestamp: 1520885107605
[INFO] Storing buildScmBranch: master
[INFO]
[INFO] --- maven-resources-plugin:3.0.1:resources (default-resources) @ wildfly-swarm ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 1 resource
[INFO]
[INFO] --- maven-compiler-plugin:3.6.0:compile (default-compile) @ wildfly-swarm ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 14 source files to /var/folders/dd/gx1cd7ws7wq8tf6phrjxqbmsnnzm5j/T/tmpdir1109678367441608430/target/classes
[INFO]
[INFO] --- maven-resources-plugin:3.0.1:testResources (default-testResources) @ wildfly-swarm ---
[INFO] Not copying test resources
[INFO]
[INFO] --- maven-compiler-plugin:3.6.0:testCompile (default-testCompile) @ wildfly-swarm ---
[INFO] Not compiling test sources
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 4 source files to /var/folders/dd/gx1cd7ws7wq8tf6phrjxqbmsnnzm5j/T/tmpdir1109678367441608430/target/test-classes
[INFO]
[INFO] --- maven-surefire-plugin:2.19.1:test (default-test) @ wildfly-swarm ---
[INFO] Tests are skipped.
[INFO]
[INFO] --- maven-jar-plugin:3.0.2:jar (default-jar) @ wildfly-swarm ---
[INFO] Building jar: /var/folders/dd/gx1cd7ws7wq8tf6phrjxqbmsnnzm5j/T/tmpdir1109678367441608430/target/wildfly-swarm-addon.jar
[INFO]
[INFO] --- maven-source-plugin:3.0.1:jar-no-fork (attach-sources) @ wildfly-swarm ---
[INFO] Building jar: /var/folders/dd/gx1cd7ws7wq8tf6phrjxqbmsnnzm5j/T/tmpdir1109678367441608430/target/wildfly-swarm-addon-sources.jar
[INFO]
[INFO] --- maven-jar-plugin:3.0.2:jar (create-forge-addon) @ wildfly-swarm ---
[INFO] Building jar: /var/folders/dd/gx1cd7ws7wq8tf6phrjxqbmsnnzm5j/T/tmpdir1109678367441608430/target/wildfly-swarm-addon-forge-addon.jar
[INFO]
[INFO] --- maven-install-plugin:2.5.2:install (default-install) @ wildfly-swarm ---
[INFO] Installing /var/folders/dd/gx1cd7ws7wq8tf6phrjxqbmsnnzm5j/T/tmpdir1109678367441608430/target/wildfly-swarm-addon.jar to /Users/micheom/.m2/repository/org/jboss/forge/addon/wildfly-swarm/2018.4.0-SNAPSHOT/wildfly-swarm-2018.4.0-SNAPSHOT.jar
[INFO] Installing /var/folders/dd/gx1cd7ws7wq8tf6phrjxqbmsnnzm5j/T/tmpdir1109678367441608430/pom.xml to /Users/micheom/.m2/repository/org/jboss/forge/addon/wildfly-swarm/2018.4.0-SNAPSHOT/wildfly-swarm-2018.4.0-SNAPSHOT.pom
[INFO] Installing /var/folders/dd/gx1cd7ws7wq8tf6phrjxqbmsnnzm5j/T/tmpdir1109678367441608430/target/wildfly-swarm-addon-sources.jar to /Users/micheom/.m2/repository/org/jboss/forge/addon/wildfly-swarm/2018.4.0-SNAPSHOT/wildfly-swarm-2018.4.0-SNAPSHOT-sources.jar
[INFO] Installing /var/folders/dd/gx1cd7ws7wq8tf6phrjxqbmsnnzm5j/T/tmpdir1109678367441608430/target/wildfly-swarm-addon-forge-addon.jar to /Users/micheom/.m2/repository/org/jboss/forge/addon/wildfly-swarm/2018.4.0-SNAPSHOT/wildfly-swarm-2018.4.0-SNAPSHOT-forge-addon.jar
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 3.057 s
[INFO] Finished at: 2018-03-12T15:05:09-05:00
[INFO] Final Memory: 81M/1064M
[INFO] ------------------------------------------------------------------------
***INFO*** Installing Addon from Git:Removing previous addon installation (org.jboss.forge.addon:wildfly-swarm,2018.4.0-SNAPSHOT) 	[3/4] ...
***INFO*** Installing Addon from Git:Installing addon (org.jboss.forge.addon:wildfly-swarm,2018.4.0-SNAPSHOT) 	[4/4] ...
***SUCCESS*** Addon(s) [org.jboss.forge.addon:wildfly-swarm,2018.4.0-SNAPSHOT] installed successfully.

**Pull request merged**